### PR TITLE
Part3

### PR DESCRIPTION
**Pull request by kstoops815**

### CODE REVIEW

**Comment by kstoops815** [comment]
@rousell If I didn't miss anything, I only need help on one part. I need to get the form validation working. I think everything else is working as stated in the instructions for part3.

File: javascripts/AppConfig.js [hunk 1]
```diff
@@ -1,11 +1,51 @@
...
+  });
 });
 
+
+
+
+
 app.config(function($routeProvider){
 	$routeProvider
+	.when("/auth", {
```
Comment (morecallan) [inline javascripts/AppConfig.js]: This should be `/login` instead of `/auth`.

**Review by morecallan — COMMENTED** [review]
@kstoops815 - Close! One small tweek + make sure you deploy. We can talk through validate if you want!

File: javascripts/services/ContactsService.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	const getAllContacts = (userUid) => {
+		let contacts = [];
+		return $q((resolve, reject) => {
+			$http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+				let fbContacts = results.data;
+				console.log("results", results);
+				Object.keys(fbContacts).forEach((key) => {
+					fbContacts[key].id = key;
+					contacts.push(fbContacts[key]);
+					resolve(contacts);
```
Comment (rousell) [inline javascripts/services/ContactsService.js]: make sure to resolve your contacts after you completed your forEach loop through the Object.keys of your fbContacts

**Review by rousell — COMMENTED** [review]
@kstoops815 sorry, one last thing! right now, if I click the X to delete the last contact for my user, it doesn't delete from the UI of the page; however, it does appear to delete successfully in firebase. Please look at my code comment on how to resolve this issue.

**Comment by rousell** [comment]
@kstoops815 great, looks good! 👍  merging in

**Pull request merged**